fabric-item-api-v1/src/main/java/net/fabricmc/fabric/impl/item/FabricItemPredicateCodec.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.item;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.ImmutableList;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.DynamicOps;
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.MapLike;
+import com.mojang.serialization.RecordBuilder;
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.predicate.item.ItemPredicate;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.dynamic.RuntimeOps;
+
+import net.fabricmc.fabric.api.item.v1.CustomItemPredicate;
+
+public final class FabricItemPredicateCodec extends MapCodec<ItemPredicate> {
+	public static final BiMap<Identifier, Codec<CustomItemPredicate>> REGISTRY = HashBiMap.create();
```

Comment:
Shouldn't this be in a separate class? e.g. `CustomItemPredicateRegistryImpl`. Just to maintain consistency + not mix registration logic and codec logic

Reply:
Not worth creating a separate class for one field IMO.